diff --git a/src/widgets/slider.rs b/src/widgets/slider.rs
index b5ce4fc..c77ce18 100644
--- a/src/widgets/slider.rs
+++ b/src/widgets/slider.rs
@@ -63,6 +63,7 @@ pub enum SliderClamping {
     /// or dragging the number next to the slider will always work.
     ///
     /// The actual slider part is always clamped though.
+    #[default]
     Never,
 
     /// Users cannot enter new values that are outside the range.
@@ -71,15 +72,14 @@ pub enum SliderClamping {
     Edits,
 
     /// Always clamp values, even existing ones.
-    #[default]
     Always,
 }
 
 /// Control a number with a slider.
 ///
 /// The slider range defines the values you get when pulling the slider to the far edges.
-/// By default all values are clamped to this range, even when not interacted with.
-/// You can change this behavior by passing `false` to [`Slider::clamping`].
+/// By default, dragging the slider is constrained to the range, while values entered in the
+/// value editor are unrestricted. You can change this behavior with [`Slider::clamping`].
 ///
 /// The range can include any numbers, and go from low-to-high or from high-to-low.
 ///
@@ -123,7 +123,7 @@ pub struct Slider<'a> {
 impl<'a> Slider<'a> {
     /// Creates a new horizontal slider.
     ///
-    /// The `value` given will be clamped to the `range`,
+    /// The slider track is constrained to `range`, but the `value` and text edits are unrestricted
     /// unless you change this behavior with [`Self::clamping`].
     pub fn new<Num: emath::Numeric>(
         value: &'a mut Num,
@@ -251,7 +251,7 @@ impl<'a> Slider<'a> {
 
     /// Controls when the values will be clamped to the range.
     ///
-    /// ### With `.clamping(SliderClamping::Always)` (default)
+    /// ### With `.clamping(SliderClamping::Always)`
     /// ```
     /// # egui::__run_test_ui(|ui| {
     /// let mut my_value: f32 = 1337.0;
@@ -275,7 +275,7 @@ impl<'a> Slider<'a> {
     /// # });
     /// ```
     ///
-    /// ### With `.clamping(SliderClamping::Never)`
+    /// ### With `.clamping(SliderClamping::Never)` (default)
     /// ```
     /// # egui::__run_test_ui(|ui| {
     /// let mut my_value: f32 = 1337.0;
@@ -1208,3 +1208,106 @@ fn logarithmic_zero_cutoff(min: f64, max: f64) -> f64 {
     );
     cutoff
 }
+
+pub struct CustomSlider<'a>(Slider<'a>);
+impl<'a> Slider<'a> {
+    pub fn custom(self) -> CustomSlider<'a> {
+        CustomSlider(self)
+    }
+}
+impl<'a> CustomSlider<'a> {
+    fn add_contents(&mut self, ui: &mut Ui) -> Response {
+        let slider = &mut self.0;
+
+        let old_value = slider.get_value();
+
+        if slider.clamping == SliderClamping::Always {
+            slider.set_value(old_value);
+        }
+
+        let thickness = ui
+            .text_style_height(&TextStyle::Body)
+            .at_least(ui.spacing().interact_size.y);
+        let mut response = slider.allocate_slider_space(ui, thickness);
+        slider.slider_ui(ui, &response);
+
+        let value = slider.get_value();
+        if value != old_value {
+            response.mark_changed();
+        }
+        response.widget_info(|| WidgetInfo::slider(ui.is_enabled(), value, slider.text.text()));
+
+        ui.ctx().accesskit_node_builder(response.id, |builder| {
+            use accesskit::Action;
+            builder.set_min_numeric_value(*slider.range.start());
+            builder.set_max_numeric_value(*slider.range.end());
+            if let Some(step) = slider.step {
+                builder.set_numeric_value_step(step);
+            }
+            builder.add_action(Action::SetValue);
+
+            let clamp_range = if slider.clamping == SliderClamping::Never {
+                f64::NEG_INFINITY..=f64::INFINITY
+            } else {
+                slider.range()
+            };
+            if value < *clamp_range.end() {
+                builder.add_action(Action::Increment);
+            }
+            if value > *clamp_range.start() {
+                builder.add_action(Action::Decrement);
+            }
+        });
+
+
+        let slider_response = response.clone();
+        
+        ui.horizontal(|ui| {
+            let label_response = ui.add(Label::new(slider.text.clone()).extend());
+            let value_response = if slider.show_value {
+                let handle_shape = slider
+                    .handle_shape
+                    .unwrap_or_else(|| ui.style().visuals.handle_shape);
+                let position_range = slider.position_range(&response.rect, &handle_shape);
+                let value_response = slider.value_ui(ui, position_range);
+                if value_response.gained_focus()
+                    || value_response.has_focus()
+                    || value_response.lost_focus()
+                {
+                    // Use the [`DragValue`] id as the id of the whole widget,
+                    // so that the focus events work as expected.
+                    response = value_response.union(response);
+                } else {
+                    // Use the slider id as the id for the whole widget
+                    response = response.union(value_response.clone());
+                }
+                Some(value_response)
+            } else {
+                None
+            };
+            if !slider.text.is_empty() {
+                // The slider already has an accessibility label via widget info,
+                // but sometimes it's useful for a screen reader to know
+                // that a piece of text is a label for another widget,
+                // e.g. so the text itself can be excluded from navigation.
+                slider_response.labelled_by(label_response.id);
+                if let Some(value_response) = value_response {
+                    value_response.labelled_by(label_response.id);
+                }
+            }
+            response
+        }).inner
+    }
+}
+impl<'a> Widget for CustomSlider<'a> {
+    fn ui(mut self, ui: &mut Ui) -> Response {
+        let inner_response = crate::Frame::group(ui.style())
+            .outer_margin(crate::Margin::symmetric(0, 2))
+            .show(ui, |ui| {
+                ui.allocate_ui_with_layout(vec2(0.0, 40.0), crate::Layout::bottom_up(crate::Align::Min), |ui| {
+                    self.add_contents(ui)
+                })
+            });
+        inner_response.inner.inner | inner_response.inner.response | inner_response.response
+    }
+}
